### PR TITLE
Reuse grapher checkboxes in site

### PR DESCRIPTION
**Pull request by mlbrgl**
Screenshot 2023-11-02 at 19 15 43

For #2890, we need a redesigned checkbox. 

- this PR makes the recently redesigned checkbox available to the site context.

- I'm adding the capability to render nodes in addition to strings in the label. This is to cater for links in the label ("list of donors")

- there is a [global grapher style](https://github.com/owid/owid-grapher/blob/8ad476af1ab409f828fb364898d58574a9a1870f/packages/%40ourworldindata/grapher/src/core/grapher.scss#L118-L120) which takes care of the checkbox hover state. This style is not available in the site context. I chose to add it to the component itself to make it self-contained. There are other places which rely on the global style, and would require a deeper refactor if we wanted to get rid of it. Here is a demo of the problem this fixes:

https://github.com/owid/owid-grapher/assets/13406362/8ff41bac-bb43-40d5-95bc-ce6674f3fa1d

### CODE REVIEW

**Comment by mlbrgl** [comment]
Current dependencies on/for this PR:
* master
  * **PR #2888** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/2888" target="_blank">Graphite</a>  👈
    * **PR #2889** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/2889" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/owid/owid-grapher/2888?utm_source=stack-comment).

**Comment by mlbrgl** [comment]
@danyx23 you recently created a `components` package. Should the checkbox tsx and scss be moved there?

File: packages/@ourworldindata/grapher/src/controls/Checkbox.tsx [hunk 1]
```diff
@@ -4,24 +4,26 @@ import { faCheck } from "@fortawesome/free-solid-svg-icons"
 
 export class Checkbox extends React.Component<{
     checked: boolean
-    label: string
     onChange: () => any
+    label: string | (() => React.ReactNode)
```
Comment (sophiamersmann) [inline packages/@ourworldindata/grapher/src/controls/Checkbox.tsx:8]: out of interest, why is the label a function and not a plain react node?

**Review by sophiamersmann — APPROVED** [review]
Looks good to me!

I think it would make sense to move the checkbox into the `components` package

Agreed that `grapher.scss` should be cleaned up :)

File: packages/@ourworldindata/grapher/src/controls/Checkbox.tsx (hunk 1 above)
Comment (mlbrgl) [reply packages/@ourworldindata/grapher/src/controls/Checkbox.tsx]: thanks @sophiamersmann!

Yes you're right, we don't need the render prop here 👍

**Comment by mlbrgl** [comment]
### Merge activity

* **Nov 9, 3:32 AM**: @mlbrgl started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/owid/owid-grapher/2888).

**Pull request merged**